src/v/cluster/rm_stm.h

```diff
@@ -369,6 +371,9 @@ class rm_stm final : public persisted_stm {
         // a heap of the first offsets of the ongoing transactions
         absl::btree_set<model::offset> ongoing_set;
         absl::flat_hash_map<model::producer_identity, prepare_marker> prepared;
+        // After transaction GA we do not write prepare marker, so we need to
```

Comment:
I think comments like "After transaction GA" don't age well (reader won't know what/when that is in a couple of months), probably replace with some context here or totally omit it.